HDDS-1986. Fix listkeys API.

bharatviswa504 opened the pull request:
Implement listKeys API.

smengcl commented:
/label ozone

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 1 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 38 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 39 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-hdds in trunk failed. |
| -1 | compile | 12 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 63 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 845 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 21 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 19 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 944 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 34 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 20 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 36 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 40 | hadoop-ozone in the patch failed. |
| -1 | compile | 22 | hadoop-hdds in the patch failed. |
| -1 | compile | 17 | hadoop-ozone in the patch failed. |
| -1 | javac | 22 | hadoop-hdds in the patch failed. |
| -1 | javac | 17 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 31 | hadoop-ozone: The patch generated 4 new + 0 unchanged - 0 fixed = 4 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 709 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 21 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 19 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 31 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 18 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 29 | hadoop-hdds in the patch failed. |
| -1 | unit | 27 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 34 | The patch does not generate ASF License warnings. |
| | | 2342 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1588 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux dbf2530a1ece 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 76605f1 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/testReport/ |
| Max. process+thread count | 441 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 42 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 33 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 31 | hadoop-ozone in trunk failed. |
| -1 | compile | 22 | hadoop-hdds in trunk failed. |
| -1 | compile | 16 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 54 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 823 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 20 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 22 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 920 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 31 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 19 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 35 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 39 | hadoop-ozone in the patch failed. |
| -1 | compile | 24 | hadoop-hdds in the patch failed. |
| -1 | compile | 20 | hadoop-ozone in the patch failed. |
| -1 | javac | 24 | hadoop-hdds in the patch failed. |
| -1 | javac | 20 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 32 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 695 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 20 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 20 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 32 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 22 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 28 | hadoop-hdds in the patch failed. |
| -1 | unit | 26 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 2305 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1588 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 09b9504ba0c3 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 76605f1 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/testReport/ |
| Max. process+thread count | 441 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 76 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 29 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 32 | hadoop-ozone in trunk failed. |
| -1 | compile | 19 | hadoop-hdds in trunk failed. |
| -1 | compile | 13 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 48 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 930 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 18 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 16 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 1024 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 34 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 21 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 35 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 36 | hadoop-ozone in the patch failed. |
| -1 | compile | 22 | hadoop-hdds in the patch failed. |
| -1 | compile | 17 | hadoop-ozone in the patch failed. |
| -1 | javac | 22 | hadoop-hdds in the patch failed. |
| -1 | javac | 17 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 28 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 787 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 25 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 21 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 32 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 18 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 26 | hadoop-hdds in the patch failed. |
| -1 | unit | 25 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 2495 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1588 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 80553b9dfed3 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a3cf54c |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/testReport/ |
| Max. process+thread count | 305 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/3/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 32 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 32 | hadoop-ozone in trunk failed. |
| -1 | compile | 22 | hadoop-hdds in trunk failed. |
| -1 | compile | 16 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 50 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 843 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 22 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 20 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 943 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 32 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 21 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 34 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 38 | hadoop-ozone in the patch failed. |
| -1 | compile | 25 | hadoop-hdds in the patch failed. |
| -1 | compile | 31 | hadoop-ozone in the patch failed. |
| -1 | javac | 25 | hadoop-hdds in the patch failed. |
| -1 | javac | 31 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 41 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 735 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 22 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 21 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 33 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 21 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 27 | hadoop-hdds in the patch failed. |
| -1 | unit | 28 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 2387 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1588 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux eb9f4e7930dc 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / f209722 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/4/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:
```diff
@@ -680,26 +688,85 @@ public boolean isBucketEmpty(String volume, String bucket)
+    // of maxKeys from the sorted map.
+    currentCount = 0;
+
+    for (Map.Entry<String, OmKeyInfo>  cacheKey : cacheKeyMap.entrySet()) {
```
The second iteration is unfortunate. We should see if there is a way to avoid it.

arp7 commented on hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestOmMetadataManager.java:
```diff
@@ -0,0 +1,298 @@
+  }
+
+  @Test
+  public void testListKeysWithFewDeleteEntriesInCache() throws Exception {
```
How are you ensuring entries are in the cache? For that you have to pause double buffer flush right?

anuengineer commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:
```diff
@@ -680,26 +688,85 @@ public boolean isBucketEmpty(String volume, String bucket)
+    // and construct treeMap which match with keyPrefix and are greater than or
+    // equal to startKey. Later we can revisit this, if list operation
+    // is becoming slow.
+    while (iterator.hasNext()) {
```
I am ok with putting this change in if we can prove that we can do large list keys. You might want to borrow the DB from @nandakumar131 and see if you can list keys with this patch, just a thought.

bharatviswa504 replied:
The key cache is not full cache, so if double buffer flush is going on well in background, this should have around couple of 100 entries. When I started freon with 10 threads, i see the value of maximum iteration is 200. So, almost in the cache we have 200 entries. (But on tried with busy workload clusters, slow disks)

arp7 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:
```diff
@@ -645,7 +648,12 @@ public boolean isBucketEmpty(String volume, String bucket)
       String startKey, String keyPrefix, int maxKeys) throws IOException {
+
     List<OmKeyInfo> result = new ArrayList<>();
+    if (maxKeys == 0) {
```
Yeah that would be a nice bit of defensive programming. Let's make the check `<= 0`.

bharatviswa504 commented:
Addressed the review comment.

bharatviswa504 commented:
> +1
> 
> I am okay to get this committed with a minor comment below, assuming there are no unaddressed comments from @anuengineer.
> 
> We should benchmark list operations later in case any further optimization is needed.

I will run the benchmarks once after the list Operations is fixed.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 46 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 36 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 40 | hadoop-ozone in trunk failed. |
| -1 | compile | 23 | hadoop-hdds in trunk failed. |
| -1 | compile | 15 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 60 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 858 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 23 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 20 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 959 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 34 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 20 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 36 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 36 | hadoop-ozone in the patch failed. |
| -1 | compile | 25 | hadoop-hdds in the patch failed. |
| -1 | compile | 20 | hadoop-ozone in the patch failed. |
| -1 | javac | 25 | hadoop-hdds in the patch failed. |
| -1 | javac | 20 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 30 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 720 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 23 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 19 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 30 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 22 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 29 | hadoop-hdds in the patch failed. |
| -1 | unit | 26 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 2387 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.3 Server=19.03.3 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1588 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 340394fd7c3d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / eeb58a0 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/5/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

anuengineer commented:
Let us get this in, I expect some of these things we will learn the right choices only when we really benchmark and test. The sad this is that some of these changes can make our system unstable. 
FYI: @elek , I know that you might not be happy with this approach. But it is hard to judge the impact of these change till we have this code in and start testing.

bharatviswa504 commented:
Thank You @arp7 and @anuengineer for the review.
I will try to run the benchmark and update it.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 1384 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 30 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 34 | hadoop-ozone in trunk failed. |
| -1 | compile | 22 | hadoop-hdds in trunk failed. |
| -1 | compile | 16 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 51 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 864 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 24 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 21 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 965 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 30 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 21 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 36 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 37 | hadoop-ozone in the patch failed. |
| -1 | compile | 26 | hadoop-hdds in the patch failed. |
| -1 | compile | 20 | hadoop-ozone in the patch failed. |
| -1 | javac | 26 | hadoop-hdds in the patch failed. |
| -1 | javac | 20 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 57 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 713 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 23 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 19 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 32 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 20 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 28 | hadoop-hdds in the patch failed. |
| -1 | unit | 26 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 34 | The patch does not generate ASF License warnings. |
| | | 3711 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.3 Server=19.03.3 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1588 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 12ab8a1f7489 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4850b3a |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/testReport/ |
| Max. process+thread count | 441 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1588/6/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

nandakumar131 commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:
```diff
@@ -680,26 +688,85 @@ public boolean isBucketEmpty(String volume, String bucket)
+      // Making sure that entry in cache is not for delete key request.
+
+      if (omKeyInfo != null) {
+        if (key.startsWith(seekPrefix) && key.compareTo(seekKey) >= 0) {
```
If there is a key which matches exactly with the `keyPrefix` that is passed to the list call and the `startKey` value is null/empty, we have to include that in the result.
If that particular key happens to be in the Cache, we are ignoring it.
`key.compareTo(seekKey) >= 0` --> will evaluate to false

bharatviswa504 replied:
if keyPrefix is null, we consider keyPrefix=/vol/bucket/

nandakumar131 replied:
` if (StringUtil.isNotBlank(startKey)) {
      // Seek to the specified key.
      seekKey = getOzoneKey(volumeName, bucketName, startKey);
      skipStartKey = true;
    } else {
      // This allows us to seek directly to the first key with the right prefix.
      seekKey = getOzoneKey(volumeName, bucketName, keyPrefix);
    }`

nandakumar131 replied:
``` if (StringUtil.isNotBlank(startKey)) {
      // Seek to the specified key.
      seekKey = getOzoneKey(volumeName, bucketName, startKey);
      skipStartKey = true;
    } else {
      // This allows us to seek directly to the first key with the right prefix.
      seekKey = getOzoneKey(volumeName, bucketName, keyPrefix);
    }```

nandakumar131 replied:
```
  if (StringUtil.isNotBlank(startKey)) {
      // Seek to the specified key.
      seekKey = getOzoneKey(volumeName, bucketName, startKey);
      skipStartKey = true;
    } else {
      // This allows us to seek directly to the first key with the right prefix.
      seekKey = getOzoneKey(volumeName, bucketName, keyPrefix);
    }
```

bharatviswa504 replied:
Let's take an example: we want to list all keys in /vol/buck with startKey=null, keyPrefix="key2"

/vol/buck/key1
/vol/buck/key2
/vol/buck/key3

keyPrefix="key2"
startKey=null

so now seeKey="/vol/buck/" seekPrefix="/vol/buck/key2"
so now when we iterate, we add to map, /vol/buck/key2 because it starts with seekKey and key.compareTo will return zero.
similarly /vol/buck/key3

then when we finally return we will only return /vol/buck/key2 and /vol/buck/key3.

Not sure if something i am missing here.

bharatviswa504 replied:
finally merge
for (Map.Entry<String, OmKeyInfo>  cacheKey : cacheKeyMap.entrySet()) {
      if (cacheKey.getKey().equals(seekKey) && skipStartKey) {
        continue;
      }

      result.add(cacheKey.getValue());
      currentCount++;

      if (currentCount == maxKeys) {
        break;
      }
    }
we have taken care right?

nandakumar131 replied:
Yup, thanks for the explanation.

bharatviswa504 commented:
Thank You all for the review.
I have committed this to the trunk.
